File: backend/src/main/java/com/stocks/sample/repository/StockRepository.java
package com.stocks.sample.repository;

import com.stocks.sample.domain.Stock;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface StockRepository extends JpaRepository<Stock, Long> {

    @Query("select s from Stock s where s.symbol = ?1")
    List<Stock> findBySymbol(String symbol);

    @Query("select s from Stock s where upper(s.symbol) like upper(?1) or upper(s.name) like upper(?1) order by s.name asc")
    List<Stock> findByWildCard(String symbol);
}

File: backend-dotnet/BackendDotNet/src/BackendDotNet.Library/Repository/StockRepository.cs
using System;
using BackendDotNet.Common.NHibernate;
using BackendDotNet.Domain;
using NHibernate;
using System.Collections.Generic;

namespace BackendDotNet.Repository {
	public class StockRepository : Repository<long, Stock> {
		public StockRepository (IUow uow) : base (uow) {
		}

		public IList<Stock> FindByWildCard( string phrase ) {
			string query = String.Format( "select s from Stock s where upper(s.Symbol) like upper('{0}') or upper(s.Name) like upper('{0}') order by s.Name asc", phrase );
			return this.Uow.Session.CreateQuery ( query ).List<Stock> ();
		}
	}
}